core/lib/multivm/src/versions/vm_fast/tests/code_oracle.rs

```diff
@@ -147,3 +159,80 @@ fn test_code_oracle_big_bytecode() {
     let result = vm.vm.execute(VmExecutionMode::OneTx);
     assert!(!result.result.is_failed(), "Transaction wasn't successful");
 }
+
+#[test]
+fn refunds_in_code_oracle() {
+    let precompiles_contract_address = Address::random();
+    let precompile_contract_bytecode = read_precompiles_contract();
+
+    let normal_zkevm_bytecode = read_test_contract();
+    let normal_zkevm_bytecode_hash = hash_bytecode(&normal_zkevm_bytecode);
+    let normal_zkevm_bytecode_keccak_hash = keccak256(&normal_zkevm_bytecode);
+    let mut storage = get_empty_storage();
+    storage.set_value(
+        get_known_code_key(&normal_zkevm_bytecode_hash),
+        u256_to_h256(U256::one()),
+    );
+
+    let precompile_contract = load_precompiles_contract();
+    let call_code_oracle_function = precompile_contract.function("callCodeOracle").unwrap();
+
+    // Execute code oracle twice with identical VM state that only differs in that the queried bytecode
+    // is already decommitted the second time. The second call must consume less gas (`decommit` doesn't charge additional gas
+    // for already decommitted codes).
+    let mut oracle_costs = vec![];
+    for decommit in [false, true] {
+        let mut vm = VmTesterBuilder::new()
+            .with_execution_mode(TxExecutionMode::VerifyExecute)
+            .with_random_rich_accounts(1)
+            .with_custom_contracts(vec![(
+                precompile_contract_bytecode.clone(),
+                precompiles_contract_address,
+                false,
+            )])
+            .with_storage(storage.clone())
+            .build();
+
+        vm.vm.insert_bytecodes([normal_zkevm_bytecode.as_slice()]);
+
+        let account = &mut vm.rich_accounts[0];
+        if decommit {
+            let (_, is_fresh) = vm
+                .vm
+                .inner
+                .world_diff
+                .decommit_opcode(&mut vm.vm.world, h256_to_u256(normal_zkevm_bytecode_hash));
+            assert!(is_fresh);
+        }
+
+        let tx = account.get_l2_tx_for_execute(
+            Execute {
+                contract_address: precompiles_contract_address,
+                calldata: call_code_oracle_function
+                    .encode_input(&[
+                        Token::FixedBytes(normal_zkevm_bytecode_hash.0.to_vec()),
+                        Token::FixedBytes(normal_zkevm_bytecode_keccak_hash.to_vec()),
+                    ])
+                    .unwrap(),
+                value: U256::zero(),
+                factory_deps: vec![],
+            },
+            None,
+        );
+
+        vm.vm.push_transaction(tx);
+        let result = vm.vm.execute(VmExecutionMode::OneTx);
+        assert!(!result.result.is_failed(), "Transaction wasn't successful");
```

Comment:
```suggestion
        assert!(!result.result.is_failed(), "Transaction wasn't successful: {result:#?}");
```

that could save a bit of time